shared/folders/index.js

```diff
@@ -55,18 +52,17 @@ class Folders extends Component<void, Props, void> {
 }
 
 export default connect(
-  (state: TypedState) => ({
+  (state: TypedState, {routeProps, routeState}) => ({
     username: state.config.username,
     folderState: state.favorite ? state.favorite.folderState : null,
-    showingPrivate: !!state.favorite && state.favorite.viewState.showingPrivate,
-    publicShowingIgnored: !!state.favorite && state.favorite.viewState.publicIgnoredOpen,
-    privateShowingIgnored: !!state.favorite && state.favorite.viewState.privateIgnoredOpen,
+    showingPrivate: !!state.favorite && routeProps.showingPrivate,
+    showingIgnored: !!state.favorite && routeState.showingIgnored,
   }),
-  (dispatch: any) => ({
+  (dispatch: any, {routePath, routeState, setRouteState}) => ({
     favoriteList: () => { dispatch(favoriteList()) },
-    routeAppend: path => { dispatch(routeAppend(path)) },
+    navigateAppend: path => { dispatch(navigateAppend({selected: 'files', path})) },
     openInKBFS: path => { dispatch(openInKBFS(path)) },
-    switchTab: showingPrivate => { dispatch(switchTab(showingPrivate)) },
-    onToggleShowIgnored: isPrivate => { dispatch(onToggleShowIgnored(isPrivate)) },
+    switchTab: showingPrivate => { dispatch(switchTo(...routePath.slice(0, -1), showingPrivate ? 'private' : 'public')) },
```

Comment:
I think routePath should be an immutable struct, then you can do stuff that's slightly easer to understand: 
like: `routePath.pop().push(showingPrivate ? 'private' : 'public')`

or maybe it would make more sense to have this be:
`navUp(); navTo(showingPrivate ? ...)`